diff -Naur libs/boost-lib/boost/mpl/apply.hpp /opt/home/ilya/opt/programming/atom-project/Temp/bombono-dvd/bombono-dvd-0.5.1/libs/boost-lib/boost/mpl/apply.hpp
--- libs/boost-lib/boost/mpl/apply.hpp	2009-04-18 23:28:38.000000000 +0400
+++ /opt/home/ilya/opt/programming/atom-project/Temp/bombono-dvd/bombono-dvd-0.5.1/libs/boost-lib/boost/mpl/apply.hpp	2009-09-10 14:53:22.000000000 +0400
@@ -135,7 +135,10 @@
 
 ///// iteration, depth == 1
 
-#elif BOOST_PP_ITERATION_DEPTH() == 1
+// For gcc 4.4 compatability, we must include the
+// BOOST_PP_ITERATION_DEPTH test inside an #else clause.
+#else // BOOST_PP_IS_ITERATING
+#if BOOST_PP_ITERATION_DEPTH() == 1
 
 #   define i_ BOOST_PP_FRAME_ITERATION(1)
 
@@ -222,4 +225,5 @@
 
 #   undef i_
 
+#endif // BOOST_PP_ITERATION_DEPTH()
 #endif // BOOST_PP_IS_ITERATING
diff -Naur libs/boost-lib/boost/mpl/apply_wrap.hpp /opt/home/ilya/opt/programming/atom-project/Temp/bombono-dvd/bombono-dvd-0.5.1/libs/boost-lib/boost/mpl/apply_wrap.hpp
--- libs/boost-lib/boost/mpl/apply_wrap.hpp	2009-04-18 23:28:38.000000000 +0400
+++ /opt/home/ilya/opt/programming/atom-project/Temp/bombono-dvd/bombono-dvd-0.5.1/libs/boost-lib/boost/mpl/apply_wrap.hpp	2009-09-10 14:53:36.000000000 +0400
@@ -78,7 +78,10 @@
 
 ///// iteration, depth == 1
 
-#elif BOOST_PP_ITERATION_DEPTH() == 1
+// For gcc 4.4 compatability, we must include the
+// BOOST_PP_ITERATION_DEPTH test inside an #else clause.
+#else // BOOST_PP_IS_ITERATING
+#if BOOST_PP_ITERATION_DEPTH() == 1
 
 #   define i_ BOOST_PP_FRAME_ITERATION(1)
 
@@ -197,4 +200,5 @@
 
 #   undef j_
 
+#endif // BOOST_PP_ITERATION_DEPTH()
 #endif // BOOST_PP_IS_ITERATING
diff -Naur libs/boost-lib/boost/mpl/aux_/full_lambda.hpp /opt/home/ilya/opt/programming/atom-project/Temp/bombono-dvd/bombono-dvd-0.5.1/libs/boost-lib/boost/mpl/aux_/full_lambda.hpp
--- libs/boost-lib/boost/mpl/aux_/full_lambda.hpp	2009-04-18 23:28:38.000000000 +0400
+++ /opt/home/ilya/opt/programming/atom-project/Temp/bombono-dvd/bombono-dvd-0.5.1/libs/boost-lib/boost/mpl/aux_/full_lambda.hpp	2009-09-10 14:53:48.000000000 +0400
@@ -227,7 +227,10 @@
 
 ///// iteration, depth == 1
 
-#elif BOOST_PP_ITERATION_DEPTH() == 1
+// For gcc 4.4 compatability, we must include the
+// BOOST_PP_ITERATION_DEPTH test inside an #else clause.
+#else // BOOST_PP_IS_ITERATING
+#if BOOST_PP_ITERATION_DEPTH() == 1
 #define i_ BOOST_PP_FRAME_ITERATION(1)
 
 #if i_ > 0
@@ -347,4 +350,5 @@
 };
 
 #undef i_
+#endif // BOOST_PP_ITERATION_DEPTH()
 #endif // BOOST_PP_IS_ITERATING
diff -Naur libs/boost-lib/boost/mpl/bind.hpp /opt/home/ilya/opt/programming/atom-project/Temp/bombono-dvd/bombono-dvd-0.5.1/libs/boost-lib/boost/mpl/bind.hpp
--- libs/boost-lib/boost/mpl/bind.hpp	2009-04-18 23:28:38.000000000 +0400
+++ /opt/home/ilya/opt/programming/atom-project/Temp/bombono-dvd/bombono-dvd-0.5.1/libs/boost-lib/boost/mpl/bind.hpp	2009-09-10 14:53:56.000000000 +0400
@@ -361,7 +361,10 @@
 
 ///// iteration, depth == 1
 
-#elif BOOST_PP_ITERATION_DEPTH() == 1
+// For gcc 4.4 compatability, we must include the
+// BOOST_PP_ITERATION_DEPTH test inside an #else clause.
+#else // BOOST_PP_IS_ITERATING
+#if BOOST_PP_ITERATION_DEPTH() == 1
 
 #   define i_ BOOST_PP_FRAME_ITERATION(1)
 
@@ -544,4 +547,5 @@
 #   endif
 #   undef j_
 
+#endif // BOOST_PP_ITERATION_DEPTH()
 #endif // BOOST_PP_IS_ITERATING
