--- libs/boost-lib/boost/lambda/core.hpp	2004-07-27 07:43:27.000000000 +0400
+++ libs/boost-lib/boost/lambda/core.hpp_pch	2007-07-06 18:16:19.000000000 +0400
@@ -61,14 +61,16 @@
 
 namespace {
 
+  // Muravj'ov - have to declare vars as static to compatilbe with gcc' precompiled
+  // headers
   // These are constants types and need to be initialised
-  boost::lambda::placeholder1_type free1 = boost::lambda::placeholder1_type();
-  boost::lambda::placeholder2_type free2 = boost::lambda::placeholder2_type();
-  boost::lambda::placeholder3_type free3 = boost::lambda::placeholder3_type();
+  static boost::lambda::placeholder1_type free1 = boost::lambda::placeholder1_type();
+  static boost::lambda::placeholder2_type free2 = boost::lambda::placeholder2_type();
+  static boost::lambda::placeholder3_type free3 = boost::lambda::placeholder3_type();
 
-  boost::lambda::placeholder1_type& _1 = free1;
-  boost::lambda::placeholder2_type& _2 = free2;
-  boost::lambda::placeholder3_type& _3 = free3;
+  static boost::lambda::placeholder1_type& _1 = free1;
+  static boost::lambda::placeholder2_type& _2 = free2;
+  static boost::lambda::placeholder3_type& _3 = free3;
   // _1, _2, ... naming scheme by Peter Dimov
 } // unnamed
    
